### PR TITLE
min_size_per_dim  + changes for scalars inputs+outputs support

### PR DESCRIPTION
**Pull request by YoelShoshan**
by allowing min_size_per_dim to be longer than the number of dims in a tensor, it makes usage easier.

for example, 

min_size_per_dim=(10_000, -1)
would then work for tensors of single dim - e.g. of size (1234, )
and also for tensors of 2(+) dims - e.g. of size (4321, 100)

this is much more nice for the user in several cases.

also - few changes for scalars inputs+outputs

### CODE REVIEW

**Review by mosheraboh — APPROVED** [review]
LGTM

File: fuse/data/ops/ops_cast.py [hunk 1]
```diff
@@ -37,12 +37,15 @@ def to_tensor(
         value: Any,
         dtype: Optional[torch.dtype] = None,
         device: Optional[torch.device] = None,
+        ignore_None: bool = False,
```
Comment (mosheraboh) [inline fuse/data/ops/ops_cast.py:40]: ignore_none

File: fuse/eval/metrics/libs/stat.py [hunk 2]
```diff
@@ -20,12 +20,22 @@ def pearson_correlation(
         :param target: target values
         :param mask: optional boolean mask. if it is provided, the metric will be applied only to the masked samples
         """
+        assert len(pred) > 0
         if isinstance(pred, Sequence):
-            pred = np.array(pred)
+            if np.isscalar(pred[0]):
+                pred = np.array(pred)
+            else:
+                pred = np.concatenate(pred)
```
Comment (mosheraboh) [inline fuse/eval/metrics/libs/stat.py:28]: here pred is numpy? necessarily same dims?

**Review by mosheraboh — DISMISSED** [review]
Looks good.
Minor comments inline.

File: fuse/eval/metrics/libs/stat.py (hunk 2 above)
Comment (YoelShoshan) [reply fuse/eval/metrics/libs/stat.py:28]: yes, in the else it's numpy, and it not necessarily same dims (it's 1d each)

**Pull request merged**